Campus-iOS/Map/MainView/MapContent.swift

```diff
@@ -91,21 +98,20 @@ struct MapContent: UIViewRepresentable {
     }
     
     func focusOnCanteen(mapView: MKMapView) {
-        if selectedCanteenName != "" {
+        if selectedCanteen != nil && selectedCanteen != focusedCanteen, let canteen = selectedCanteen {
             for i in mapView.annotations {
-                if i.title == selectedCanteenName {
+                if i.title == canteen.title {
                     let locValue: CLLocationCoordinate2D = i.coordinate
                     
                     let coordinate = CLLocationCoordinate2D(
                         latitude: locValue.latitude, longitude: locValue.longitude)
                     let span = MKCoordinateSpan(latitudeDelta: 0.01, longitudeDelta: 0.01)
                     let region = MKCoordinateRegion(center: coordinate, span: span)
                     
-                    mapView.setRegion(region, animated: true)
-                    for i in mapView.annotations {
-                        if i.title == selectedCanteenName {
-                            mapView.selectAnnotation(i, animated: true)
-                        }
+                    DispatchQueue.main.async {
```

Comment:
Is there a better way to do that?

Reply:
Actually, I really don't know, but it was also implemented with the DispatchQueue before.
If you have a better idea, feel free to let me know.

Reply:
Why is the async DispatchQueue actually required here? I'm not really getting it 😅